test/unit/lifecycle/properties.js

```diff
@@ -104,6 +104,39 @@ describe('lifecycle/property', () => {
         expect(elem.getAttribute('test-name')).to.equal('something else');
       });
 
+      it('setting the attribute updates the property value', done => {
```

Comment:
For posterity: The issue has to do with the way the property `set` and the `attributeChangedCallback` callback interact, so it doesn't quite fit in either location - leaving it here should be OK.